vowpalwabbit/search.cc

```diff
@@ -1730,11 +1725,11 @@ action search_predict(search_private& priv, example* ecs, size_t ec_cnt, ptag my
         size_t label_size = priv.is_ldf ? sizeof(CS::label) : sizeof(MC::label_t);
         void (*label_copy_fn)(void*, void*) = priv.is_ldf ? CS::cs_label.copy_label : nullptr;
 
-        ensure_size(priv.learn_ec_copy, ec_cnt);
+        priv.learn_ec_copy.resize(ec_cnt);
         for (size_t i = 0; i < ec_cnt; i++)
-          VW::copy_example_data(priv.all->audit, priv.learn_ec_copy.begin() + i, ecs + i, label_size, label_copy_fn);
+          VW::copy_example_data(priv.all->audit, priv.learn_ec_copy.data() + i, ecs + i, label_size, label_copy_fn);
```

Comment:
since `priv.learn_ec_copy` is now a vector we could potentially use normal vector access here instead of pointer arithmetic (i.e. `&priv.learn_ec_copy.at(i)` instead)